modules/embeddingsspace/src/main/java/com/theagilemonkeys/ellmental/embeddingsspace/EmbeddingsSpaceComponent.java

```
@@ -0,0 +1,69 @@
+package com.theagilemonkeys.ellmental.embeddingsspace;
+
+import com.theagilemonkeys.ellmental.core.schema.Embedding;
+import com.theagilemonkeys.ellmental.embeddingsgeneration.EmbeddingsGenerationModel;
+import com.theagilemonkeys.ellmental.embeddingsstore.EmbeddingsStore;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class EmbeddingsSpaceComponent {
+
+    private final EmbeddingsGenerationModel model;
+    private final EmbeddingsStore store;
+
+    public EmbeddingsSpaceComponent(EmbeddingsGenerationModel model, EmbeddingsStore store) {
+        this.model = model;
+        this.store = store;
+    }
+
+    public Embedding generate(String text) {
+        Embedding embedding = null;
+        try {
+            embedding = model.generateEmbedding(text);
+        } catch (RuntimeException e) {
+            // TODO: Create and catch EmbeddingGenerationException?
+            System.out.printf("Error while generating embedding tor text %s", text);
+        }
```

I'm not sure if capturing exceptions here makes sense because capturing _any_ runtime exception can mask errors and make troubleshooting more challenging. If we want to log the error, that's a good idea, but then we should re-raise the original exception or generate a new one, wrapping the original one. Also, if Java supports log levels, these logs must probably be `debug` level to avoid stuffing production logs.

---

I'll remove the try catchs then, we can detect if this is needed in the future :P